toolkit/scripts/test_package_install.sh

```diff
@@ -0,0 +1,199 @@
+enabled=1
+priority=0
+skip_if_unavailable=True
+enabled=1
+EOF
+)
+
+# create the repo metadata file
+echo "$repo_content" | sudo tee /etc/yum.repos.d/builtpackages.repo > /dev/null
+
+echo "##[debug]Created repo file with content:"
+sudo cat /etc/yum.repos.d/builtpackages.repo
+echo ""
+
+# Add daily 3.0 repo if available
+repo_content_daily=$(cat << EOF
+[daily3]
+name=Daily 3.0 repo
+baseurl=$LKG_BASE_URL
+gpgkey=file:///etc/pki/rpm-gpg/MICROSOFT-RPM-GPG-KEY file:///etc/pki/rpm-gpg/MICROSOFT-METADATA-GPG-KEY
+gpgcheck=0
+repo_gpgcheck=0
+enabled=1
+skip_if_unavailable=True
+sslverify=0
+EOF
+)
+
+if [[ $DIST_TAG == ".azl3" ]]; then
+    echo "$repo_content_daily" | sudo tee /etc/yum.repos.d/daily3.repo > /dev/null
+    echo "##[debug]Created daily repo file with content:"
+    sudo cat /etc/yum.repos.d/daily3.repo
+    echo ""
+else
+    echo "Mariner 3.0 was not detected, so skipping daily repo setup"
+fi
+
+# update the cached binary metadata.
+sudo tdnf makecache
+
+sudo tdnf repolist --refresh
+
+PACKAGES_FAIL_INSTALL=()
+PACKAGES_FAIL_UNINSTALL=()
+
+for rpm_to_install in "${RPMS_TO_INSTALL[@]}"; do
+    path_to_rpm=$(find $RPM_DIRECTORY -name "$rpm_to_install")
+    package_name_to_install=$(rpm -qp --queryformat "%{name}" "$path_to_rpm" 2>/dev/null)
+
+    if exists_in_list "$package_name_to_install" "${PACKAGES_TO_NOT_INSTALL[@]}"; then
+        echo "##[debug]Skipping package $package_name_to_install for installation test"
+        continue
+    fi
+
+    echo "------------------------------------------------"
+    echo "About to tdnf install package '$package_name_to_install' ('$rpm_to_install')"
+    # we are using package files instead of names to include subpackages produces with %{name}-{subpackage_name}
+    if ! sudo tdnf install -y "$path_to_rpm"; then
+        PACKAGES_FAIL_INSTALL+=("$package_name_to_install")
+        # package did not install, skip uninstall test
+        echo "The following package failed to install: '$package_name_to_install'"
+        continue
+    fi
+
+    if exists_in_list "$package_name_to_install" "${PACKAGES_TO_NOT_UNINSTALL[@]}"; then
+        echo "##[debug]Skipping package $package_name_to_install for uninstallation test"
+        continue
+    fi
+
+    # tdnf cannot accept the package rpm for uninstalling, only name. So we cannot test uninstalling of subpackages
+    echo "About to tdnf remove package '$package_name_to_install'"
+    if ! sudo tdnf remove -y "$package_name_to_install"; then
+        PACKAGES_FAIL_UNINSTALL+=("$package_name_to_install")
+    fi
+    echo ""
+    echo ""
+done
+
+echo ""
+echo "------------------------------------------------"
+echo "-- Printing install + uninstall error details --"
+echo "------------------------------------------------"
+echo ""
+
+if [ "${#PACKAGES_FAIL_INSTALL[@]}" -gt 0 ]; then
+    echo "##vso[task.logissue type=error]Error, ${#PACKAGES_FAIL_INSTALL[@]} packages failed to install, check logs to see the list"
```

Comment:
I think its specific to pipeline. Hope it will not cause any confusions in Actions!